elements/lisk-tree/src/merkle_tree/utils.ts

```diff
@@ -210,7 +144,7 @@ export const treeSortFn = (a: number, b: number) => {
 
 export const insertNewIndex = (arr: number[], val: number) => {
 	const insertIndex = binarySearch(arr, n => treeSortFn(val, n) < 0);
-	if (arr[insertIndex] !== val) {
+	if (arr[insertIndex - 1] !== val) {
```

Comment:
mh, is there a reason why this is one off?

Reply:
I think it's returning the length? for example,
`[8, 12, 26]`, `12` returns `2`.